userspace/libsinsp/test/public_sinsp_API/interesting_syscalls.cpp

```diff
@@ -279,32 +106,34 @@ libsinsp::events::set<ppm_sc_code> expected_sinsp_state_ppm_sc_set = {
 TEST(interesting_syscalls, enforce_sinsp_state_basic)
 {
 	auto state_ppm_sc_set = libsinsp::events::sinsp_state_sc_set();
-
-	ASSERT_TRUE(expected_sinsp_state_ppm_sc_set.equals(state_ppm_sc_set));
+	ASSERT_EQ(expected_sinsp_state_ppm_sc_set, state_ppm_sc_set);
```

Comment:
@FedeDP please also add the size check here and please confirm this fixes the issues in https://github.com/falcosecurity/libs/pull/914, both issues in the data structure and unit test setup, ty!

```suggestion
	ASSERT_EQ(expected_sinsp_state_ppm_sc_set.size(), state_ppm_sc_set.size());
```

Reply:
I will add the additional check!

Reply:
Addressed, thank you!